tests/test_interpolation.py

```diff
@@ -374,21 +565,47 @@ def test_copy_cache(restore_resolvers: Any) -> None:
 
 
 def test_clear_cache(restore_resolvers: Any) -> None:
-    OmegaConf.register_resolver("random", lambda _: random.randint(0, 10000000))
-    c = OmegaConf.create(dict(k="${random:_}"))
+    OmegaConf.new_register_resolver("random", lambda _: random.randint(0, 10000000))
+    c = OmegaConf.create(dict(k="${random:__}"))
     old = c.k
     OmegaConf.clear_cache(c)
     assert old != c.k
 
 
 def test_supported_chars() -> None:
-    supported_chars = "%_-abc123."
+    supported_chars = "abc123_/:-\\+.$%*@"
     c = OmegaConf.create(dict(dir1="${copy:" + supported_chars + "}"))
 
-    OmegaConf.register_resolver("copy", lambda x: x)
+    OmegaConf.new_register_resolver("copy", lambda x: x)
     assert c.dir1 == supported_chars
 
 
+def test_valid_key_names() -> None:
+    invalid_chars = "\\${}()[].: '\""
+    valid_chars = "".join(chr(i) for i in range(33, 128) if chr(i) not in invalid_chars)
+    cfg_dict = {valid_chars: 123, "inter": f"${{{valid_chars}}}"}
+    cfg = OmegaConf.create(cfg_dict)
+    # Test that we can access the node made of all valid characters, both
+    # directly and through interpolations.
+    assert cfg[valid_chars] == 123
+    assert cfg.inter == 123
+    # Test that all invalid characters trigger errors in interpolations.
+    for c in invalid_chars:
+        cfg_dict["invalid"] = f"${{ab{c}de}}"
+        cfg = OmegaConf.create(cfg_dict)
+        error: type
+        if c in [".", "}"]:
+            # With '.', we try to access `${ab.de}`.
+            # With "}", we try to access `${ab}`.
+            error = ConfigAttributeError
+        elif c == ":":
+            error = UnsupportedInterpolationType  # `${ab:de}`
+        else:
+            error = GrammarParseError  # other cases are all parse errors
+        with pytest.raises(error):
+            cfg.invalid
```

Comment:
Such long tests are not great.
I prefer tests that are testing one thing.
can you break this test into multiple smaller tests?

In this scenario there isn't even a shared parameterization. If there was - one thing you can do is to use a test class.

Reply:
Done in fd0a598 (split in two tests, one for valid characters and one for invalid ones, with the second one parameterized by each invalid character)

Reply:
better :)